hypershift-operator/controllers/nodepool/kubevirt/kubevirt.go

```diff
@@ -189,6 +191,20 @@ func virtualMachineTemplateBase(nodePool *hyperv1.NodePool, bootImage BootImage)
 		},
 	}
 
+	podResources := corev1.ResourceList{
+		corev1.ResourceMemory: memory,
+	}
+
+	if cores > 0 {
+		podResources[corev1.ResourceCPU] = *apiresource.NewQuantity(int64(cores), apiresource.DecimalSI)
+	}
+
+	template.Spec.Template.Spec.Domain.Resources.Requests = podResources
+
+	if guaranteedResources {
+		template.Spec.Template.Spec.Domain.Resources.Limits = podResources
+	}
+
```

Comment:
@davidvossel, this new logic cancels the previous one entirely, meaning, we now never set the domain.cpu/memory, but only the resources.limit/requested.

I wonder if this is what we want. Maybe this will be a better implementation?
```go
	if guaranteedResources {
		podResources := corev1.ResourceList{
			corev1.ResourceMemory: memory,
		}

		if cores > 0 {
			podResources[corev1.ResourceCPU] = *apiresource.NewQuantity(int64(cores), apiresource.DecimalSI)
		}

		template.Spec.Template.Spec.Domain.Resources.Requests = podResources
		template.Spec.Template.Spec.Domain.Resources.Limits = podResources
	} else {
		template.Spec.Template.Spec.Domain.Memory = &kubevirtv1.Memory{Guest: &memory}
		if cores > 0 {
			template.Spec.Template.Spec.Domain.CPU = &kubevirtv1.CPU{Cores: cores}
		}
	}
```

Or even maybe split the memory and cpu behavior, so we'll for qosClass == `"Burstable"` we'll set the guest memory, and the resource.request[cpu], and for qosClass == `"Guaranteed"`, well not set the guest memory, but only the resources field.

WDYT?

Reply:
Theoretically, only setting the CPU/MEM requests for `Burstable` should be the equivalent of using the spec.domain.CPU structs.

But for consistency with our existing behavior, i think you're right that when `Burstable` is in use, we should preserve the previous behavior and continue to set the spec.Domain.Memory and spec.Domain.CPU. My concern here is that I'm not 100% certain what the implications could be during updates if the nodePool controller results in mutating the VM spec, and that's what would occur if we started setting the resource requests even for Burstable... it woudl likely be fine, but we avoid having to consider that if we leave the setting as it was.